lib/page_parser.rb

```diff
@@ -0,0 +1,42 @@
+require 'nokogiri'
+
+module PageParser
+  class HtmlParserIncluded < HTTParty::Parser
+    def html
+      Nokogiri::HTML(body)
+    end
+  end
+
+  class Page
+    include HTTParty
+    parser HtmlParserIncluded
+  end
+
+  class LinkInfo
+    def initialize(url)
+      @url = url
+    end
+
+    def valid?
+      page.present?
+    end
+
+    def title
+      if valid?
+        return page.css('title').text.lstrip.rstrip
+      end
+    end
+
+    private
+
+    def page
+      return @page if @page
+
+      begin
+        @page = Page.get(@url)
```

Comment:
@bion I've been trying to stub `PageParser::Page.get` (I've also tried it as `PageParser::HtmlParserIncluded.get`) to respond with a local html file but haven't had any luck.  Short of that I'm having trouble thinking of any valuable way to test this module.